src/main/java/seedu/weme/ui/CommandBox.java

```diff
@@ -43,6 +55,68 @@ public CommandBox(CommandExecutor commandExecutor, CommandPrompter commandPrompt
         commandTextField.addEventFilter(KeyEvent.KEY_PRESSED, this::handleTabPress);
     }
 
+    /**
+     * Handles the key press event, {@code keyEvent}.
+     */
+    @FXML
+    private void handleKeyPress(KeyEvent keyEvent) {
+        switch (keyEvent.getCode()) {
+        case UP:
+            // As up and down buttons will alter the position of the caret,
+            // consuming it causes the caret's position to remain unchanged
+            keyEvent.consume();
+            handleLikeByKeyPress(KEYPRESS_UP);
+            break;
+        case DOWN:
+            keyEvent.consume();
+            handleLikeByKeyPress(KEYPRESS_DOWN);
+            break;
+        case LEFT:
+            keyEvent.consume();
+            handleIndexToggleByKeyPress(KEYPRESS_LEFT);
+            break;
+        case RIGHT:
+            keyEvent.consume();
+            handleIndexToggleByKeyPress(KEYPRESS_RIGHT);
+            break;
+        default:
+            // let JavaFx handle the keypress
+        }
+    }
+
+    /**
+     * Changes index in the command box with key press to enable fast liking.
+     */
+    private void handleIndexToggleByKeyPress(String keyPress) {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(commandTextField.getText().trim());
+        if (matcher.matches()) {
+            final String commandWord = matcher.group(WemeParser.COMMAND_WORD);
+            final String argument = matcher.group(WemeParser.ARGUMENTS);
+            int change = keyPress.equals(KEYPRESS_LEFT) ? -1 : 1;
+            int currentLikeCount = Integer.parseInt(argument.trim());
+            int newLikeCount = currentLikeCount + change;
+            commandTextField.setText(commandWord + " " + (newLikeCount < 1 ? 1 : newLikeCount));
+        }
+    }
+
+    /**
+     * Handles like command in the form of key press.
+     */
+    private void handleLikeByKeyPress(String keyPress) {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(commandTextField.getText().trim());
+        if (matcher.matches()) {
```

Comment:
If no match, the if block won't run and the method exits without any signal. Not sure if adding a guard statement would help readability or is just redundant.

Reply:
Interestingly, the original key event (moving the cursor to the start of line) is consumed by `keyEvent.consume()`. Will look into it to see if there's a fix.

Reply:
aha yeah i meant to add a guard statement for readability. Otherwise the entire try catch will be indented in one layer deeper, no reason for that.

Reply:
actually for this method and the method before, would it be more readable if you passed in UP and DOWN / LEFT and RIGHT instead of the entire keyevent?

Reply:
Passing in event would maintain consistency across tab, toggle and like